jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/DeploymentScanner.java

```diff
@@ -301,20 +272,56 @@ void addScannerListener(Scanner.Listener listener)
         scanner.addListener(listener);
     }
 
+    protected int getDefaultWeight(String name)
```

Comment:
I don't like this method.
We should just have `configureEnvironment(String, int)`.

Reply:
Read Greg's review, he didn't like this approach

Reply:
Well, I don't like the current one.

We decided to shy away from "default" environments, so the `core` environment was not created upfront, etc. so why now we should have upfront weights?

The weights are not everywhere, they are just in 1 method of `DeploymentScanner`. 
Only Jetty XML code should call that method, so a custom `corp-deploy` module should just be aware of Jetty's `xyz-deploy` modules, if they are used simultaneously with `corp-deploy`.

Otherwise, there are no Jetty `xyz-deploy` modules in use, we don't load our `environment-weights.properties` that we don't use, and there only will be `corp-deploy` and `spring-corp-deploy`.

Reply:
A prior commit did this 
See the revert at https://github.com/jetty/jetty.project/commit/00f95a732032e11ec18a83760adecbd223cb04fd for how it looked.

Reply:
I still don't like this approach either. I don't like the concept  of weights being seen externally at all, as it is just an abstraction that we don't need to both the user with.   I think we should just have:
```java
    void setEnvironmentOrder(List<String> environmentNames);
    List<String> getEnvironmentOrder();
```
With the default value being `["core", "static", "ee8", "ee9", "ee10", "ee11"]`. 

If internally we turn that into weights, then fine, but no weights should be in any public API.